### DISCUSSION TITLE
Linker files or not ?

### DISCUSSION THREAD

**Discussion by jnaulet** [jnaulet/OpenPicoRTOS#7] [Ideas]
It's kind of a dilemma, should picoRTOS provide linker files or not ?

On one hand, this would solve a few philosophical issues by providing a way to remove interrupt function pointers from every port, for example, adding code safety.

Ports would really be about architectures and not about BSPs, but this might be wishful thinking, as the border is blurry anyway.

On the other hand it will probably remove some flexibility regarding the BSP the user might want to use.

My take on it would be to add linker files in the devices subdirectories and see how convenient it is, or not.

Devices directories might add subdirectories for every compiler it supports, too.

This is up for debate.

**Comment by Dan6077** [comment]
As you are the maintainer, it would come down to you to keep all files up to date
Then you may have to spend time on maintaining multiple different linker files for different compiler options rather than spending time on RTOS features
As someone who is trying to understand the intricacies of the main code I am not in a place to have an opinion that would result in more work for someone else

**Reply by jnaulet** [reply]
You're right, i'm not sure i want to go down that rabbit hole. The risk is after i add the linker file i realise i need to add startup assembly code for every MCU, making the project too big for its own sake
